### PR TITLE
new: fast exit functions

### PR DESCRIPTION
**Pull request by rahuldamodar94**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -294,6 +295,25 @@ export default class Matic extends SDKClient {
...
+
+    const dataHash = this.web3Client.web3.utils.soliditySha3(
+      exitNFTId,
+      this.web3Client.web3.utils.toChecksumAddress(userAddress)
+    )
+
+    const wallet = this.web3Client.getWallet()
+    const vrs = ethUtils.ecsign(ethUtils.toBuffer(dataHash), ethUtils.toBuffer(wallet[options.from].privateKey))
+    let signature = ethUtils.toRpcSig(vrs.v, vrs.r, vrs.s)
+    return signature
```
Comment (arthcp) [inline src/index.ts:314]: We need to assume that the private key will not be available. In most of the cases, user will give metamask provider and sign things using metamask.

File: src/index.ts [hunk 2]
```diff
@@ -382,6 +402,16 @@ export default class Matic extends SDKClient {
     return this.withdrawManager.processExits(tokenAddress, options)
   }
 
+  fastExit(exitNFT: string, signature: string, fastExitAddress: string, options?: SendOptions) {
+    if (!exitNFT || !signature || !fastExitAddress) {
+      throw new Error(`txHash,signature or fastExitAddress not provided`)
+    }
+    if (options && !options.from) {
+      throw new Error(`options.from is missing`)
+    }
```
Comment (arthcp) [inline src/index.ts:411]: This condition is false if user didnt pass options object. You are using `options.from` in `safeTransferFrom` so it is always needed.

**Review by arthcp — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 3]
```diff
@@ -294,6 +295,25 @@ export default class Matic extends SDKClient {
     return this.utils.signEIP712TypedData(typedData, wallet[options.from].privateKey)
   }
 
+  async getSwapSignature(exitNFTId, userAddress, options: SendOptions) {
```
Comment (jdkanani) [inline src/index.ts:298]: Do we even need this here? This is only for fast-exits server.

**Review by jdkanani — CHANGES_REQUESTED** [review]

**Comment by jdkanani** [comment]
Do we need these functions in matic.js? I think we should keep `sig` function in fast-exits server and anyone will just call the contract directly using web3 without matic.js.

**Pull request closed without merging**